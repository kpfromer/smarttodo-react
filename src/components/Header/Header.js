import React from "react";
import PropTypes from 'prop-types';

import { connect } from 'react-redux';
import { NavLink } from "react-router-dom";
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';

import styles from './Header.module.css';

export const Header = ({ isAuthenticated }) => {

  const menuItems =
    isAuthenticated ?
      <div id="logged-in-links">
        <NavLink to="/logout">
          <Button color="inherit">
            Logout
          </Button>
        </NavLink>
      </div>
      :
      <div id="logged-out-links">
        <NavLink to="/register">
          <Button color="inherit">
            Register
          </Button>
        </NavLink>
        <NavLink to="/login">
          <Button color="inherit">
            Login
          </Button>
        </NavLink>
      </div>;

  return (
    <AppBar position="static">
      <Toolbar>
        <IconButton className={styles.menuButton} color="inherit" aria-label="Menu">
          <MenuIcon />
        </IconButton>
        <Typography variant="title" color="inherit" className={styles.flex}>
          <NavLink to="/">Smarttodo</NavLink>
        </Typography>
        {menuItems}
      </Toolbar>
    </AppBar>
  )
};

Header.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired
};

export const mapStateToProps = state => ({
  isAuthenticated: !!state.login.token
});

export default connect(mapStateToProps)(Header);