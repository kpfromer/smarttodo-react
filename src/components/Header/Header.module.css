.mainContainer {
  background-color: #3573E7;
}

.title {
  font-size: 1.35em;
  margin: 0;
}

.navList {
  margin-top: 5px;
}

.title a,
.navList a {
  text-align: center;
  display: block;
  padding: 10px 15px;
}

.navList a {
  font-size: .95em;
  text-transform: uppercase;
}

@media (min-width: 769px) {
  .navContainer,
  .navList {
    display: flex;
  }

  .navContainer {
    flex-direction: column;
    align-items: center;
  }
}

@media (min-width: 1125px) {
  .navContainer,
  .navList {
    flex-direction: row;
  }
  .navList {
    margin-left: auto;
  }
}

.flex {
  flex: 1;
}

.menuButton {
  margin-left: -12px;
  margin-right: 20px;
}