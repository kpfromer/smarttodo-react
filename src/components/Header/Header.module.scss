@import '../../theme';

.mainContainer {
  background-color: $main-color;
}

.title {
  font-size: 1.35em;
  margin: 0;
}

.navList {
  margin-top: 5px;
}

.title a,
.navList a {
  text-align: center;
  display: block;
  padding: 10px 15px;
}

.navList a {
  font-size: .95em;
  text-transform: uppercase;
}

@media (min-width: $sm) {
  .navContainer,
  .navList {
    display: flex;
  }

  .navContainer {
    flex-direction: column;
    align-items: center;
  }
}

@media (min-width: $md) {
  .navContainer,
  .navList {
    flex-direction: row;
  }
  .navList {
    margin-left: auto;
  }
}

.flex {
  flex: 1;
}

.menuButton {
  margin-left: -12px;
  margin-right: 20px;
}