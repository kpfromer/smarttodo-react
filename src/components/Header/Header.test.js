import React from "react";
import { SetupComponent } from 'react-component-setup';
import { Header } from "./Header";

const { shallow: setup } = SetupComponent({
  Component: Header,
  elementsToFind: [
    {
      name: 'title',
      query: 'h1'
    },
    {
      name: 'list',
      query: 'ul'
    }
  ]
});

describe("Header", () => {
  let items;

  beforeEach(() => {
    items = setup();
  });

  it("renders without crashing", () => {
    setup();
  });

  it('renders a title', () => {
    const { title } = items;
    expect(title).toMatchSnapshot();
  });

  it('renders a menu bar', () => {
    const { list } = items;
    expect(list).toMatchSnapshot();
  });
});