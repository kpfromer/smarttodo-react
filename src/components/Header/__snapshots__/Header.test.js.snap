// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders a menu bar 1`] = `
<ul
  className="navList"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/settings"
  >
    Settings
  </NavLink>
</ul>
`;

exports[`Header renders a title 1`] = `
<h1
  className="title"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/"
  >
    Smarttodo
  </NavLink>
</h1>
`;
